src/device-api/v1.ts

```diff
@@ -46,84 +49,50 @@ router.post('/v1/restart', (req: AuthorizedRequest, res, next) => {
-	}
-
-	return Promise.all([applicationManager.getCurrentApps(), getApp(appId)])
-		.then(([apps, targetApp]) => {
-			if (apps[appId] == null) {
-				return res.status(400).send('App not found');
-			}
-			const app = apps[appId];
-			let service = app.services[0];
-			if (service == null) {
-				return res.status(400).send('No services on app');
-			}
-			if (app.services.length > 1) {
-				return res
-					.status(400)
-					.send('Some v1 endpoints are only allowed on single-container apps');
-			}
-
-			// check that the request is scoped to cover this application
-			if (!req.auth.isScoped({ apps: [app.appId] })) {
-				return res.status(401).send('Unauthorized');
-			}
-
-			// Get the service from the target state (as we do in v2)
-			// TODO: what if we want to start a service belonging to the current app?
-			const targetService = _.find(targetApp.services, {
-				serviceName: service.serviceName,
-			});
+// QUESTION: This function is quite similar to v2's handleServiceAction
+// with the exception of some response messages being different and the presence
+// of the event tracker. Should we remove these differences and just use one function
+// between v1 & v2?
+const handleLegacyServiceAction = (action: CompositionStepAction) => {
+	return async (
+		req: AuthorizedRequest,
+		res: express.Response,
+		next: express.NextFunction,
+	) => {
+		const appId = checkInt(req.params.appId);
+		const force = checkTruthy(req.body.force);
+		eventTracker.track('Restart container (v1)', { appId });
+
+		if (appId == null) {
+			return res.status(400).send('Invalid app id');
+		}
 
-			applicationManager.setTargetVolatileForService(service.imageId, {
-				running: action !== 'stop',
-			});
+		if (!req.auth.isScoped({ apps: [appId] })) {
+			return res.status(401).send('Unauthorized');
+		}
 
-			const stopOpts = { wait: true };
-			const step = generateStep(action, {
-				current: service,
-				target: targetService,
-				...stopOpts,
+		try {
+			await actions.executeServiceAction({
+				action,
+				appId,
+				force,
 			});
-
-			return applicationManager
-				.executeStep(step, { force })
-				.then(function () {
-					if (action === 'stop') {
-						return service;
-					}
-					// We refresh the container id in case we were starting an app with no container yet
-					return applicationManager.getCurrentApps().then(function (apps2) {
-						const app2 = apps2[appId];
-						service = app2.services[0];
-						if (service == null) {
-							throw new Error('App not found after running action');
-						}
-						return service;
-					});
-				})
-				.then((service2) =>
-					res.status(200).json({ containerId: service2.containerId }),
-				);
-		})
-		.catch(next);
+			// QUESTION: The old v1 stop/start endpoints return the containerId
+			// of the new service. This 'OK' response is more in line with v2 service
+			// action responses -- should we use this or keep returning the containerId?
+			return res.status(200).send('OK');
```

Comment:
👀

Reply:
I'm not sure I'm understanding the question correctly. If the v1 interface returns the container id, then we need to keep it.

Reply:
I've added back the legacy interface in 524685d1bffc7edcf8db8cf5417f91bc08655f75, so that `POST /v1/apps/:appId/stop` and `POST /v1/apps/:appId/start` return `{ containerId: <CONTAINER_ID> }` as they did before.